packages/@aws-cdk/aws-s3/lib/rule.ts

```diff
@@ -142,6 +142,38 @@ export interface LifecycleRule {
   readonly objectSizeGreaterThan?: number;
 }
 
+/**
+ * A point in time when an object should transition to a different storage class.
+ */
+export class TransitionMoment {
+
+  /**
+   * A moment defined by a specific date.
+   *
+   * The date value must be in ISO 8601 format. The time is always midnight UTC.
+   */
+  public static at(date: Date): TransitionMoment {
+    return { transitionDate: date };
+  }
+
+  /**
+   * A moment defined by the number of days after the object creation.
+   */
+  public static after(duration: Duration): TransitionMoment {
+    return { transitionAfter: duration };
+  }
+
+  /**
+   * Indicates when objects are transitioned to the specified storage class.
+   */
+  public readonly transitionDate?: Date;
+
+  /**
+   * Indicates the number of days after creation when objects are transitioned to the specified storage class.
+   */
+  public readonly transitionAfter?: Duration;
+}
```

Comment:
why though? doesn't this just move some methods we already have onto a new class? What is the benefit we're getting by doing this?